Link error when nrnivmodl-core is used with ISPC backed

When I try to compile MOD files with ISPC backend then it fails with:

```
$ <dir>/install/bin/nrnivmodl-core lib/modlib/
...
  x86_64/core/build/_mod_func.o x86_64/core/build/_dimplic.o x86_64/core/build/Ca.o x86_64/core/build/CaDynamics_DC0.o x86_64/core/build/CaDynamics_E2.o x86_64/core/build/Ca_HVA.o x86_64/core/build/Ca_HVA2.o x86_64/core/build/Ca_LVAst.o x86_64/core/build/CoreConfig.o x86_64/core/build/Ih.o x86_64/core/build/Im.o x86_64/core/build/K_Pst.o x86_64/core/build/K_Tst.o x86_64/core/build/KdShu2007.o x86_64/core/build/NaTa_t.o x86_64/core/build/NaTg.o x86_64/core/build/NaTs2_t.o x86_64/core/build/Nap_Et2.o x86_64/core/build/ProbAMPANMDA_EMS.o x86_64/core/build/ProbGABAAB_EMS.o x86_64/core/build/ProfileHelper.o x86_64/core/build/SK_E2.o x86_64/core/build/SKv3_1.o x86_64/core/build/TTXDynamicsSwitch.o x86_64/core/build/VecStim.o x86_64/core/build/cagk.o x86_64/core/build/cal2.o x86_64/core/build/can2.o x86_64/core/build/cat.o x86_64/core/build/gap.o x86_64/core/build/h.o x86_64/core/build/halfgap.o x86_64/core/build/na3n.o x86_64/core/build/naxn.o x86_64/core/build/netstim_inhpoisson.o x86_64/core/build/new_calcium_channels.o  x86_64/core/build/Ca.obj x86_64/core/build/CaDynamics_DC0.obj x86_64/core/build/CaDynamics_E2.obj x86_64/core/build/Ca_HVA.obj x86_64/core/build/Ca_HVA2.obj x86_64/core/build/Ca_LVAst.obj x86_64/core/build/Ih.obj x86_64/core/build/Im.obj x86_64/core/build/K_Pst.obj x86_64/core/build/K_Tst.obj x86_64/core/build/KdShu2007.obj x86_64/core/build/NaTa_t.obj x86_64/core/build/NaTg.obj x86_64/core/build/NaTs2_t.obj x86_64/core/build/Nap_Et2.obj x86_64/core/build/ProbAMPANMDA_EMS.obj x86_64/core/build/ProbGABAAB_EMS.obj x86_64/core/build/SK_E2.obj x86_64/core/build/SKv3_1.obj x86_64/core/build/TTXDynamicsSwitch.obj x86_64/core/build/cagk.obj x86_64/core/build/cal2.obj x86_64/core/build/can2.obj x86_64/core/build/cat.obj x86_64/core/build/gap.obj x86_64/core/build/h.obj x86_64/core/build/halfgap.obj x86_64/core/build/na3n.obj x86_64/core/build/naxn.obj x86_64/core/build/new_calcium_channels.obj  /gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/HPCTM-1344/coreneuron/build/install/lib/libscopmath.a -L/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/HPCTM-1344/coreneuron/build/install/lib -lcoreneuron -Wl,-rpath,/gpfs/bbp.cscs.ch/project/proj16/kumbhar/pramod_scratch/HPCTM-1344/coreneuron/build/install/lib   -lpthread /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/tools/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/hpe-mpi-2.21-7pbszh6v5u/lib/libmpi++.so /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/tools/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/hpe-mpi-2.21-7pbszh6v5u/lib/libmpi.so
ld: x86_64/core/build/Ca.obj: relocation R_X86_64_PC32 against symbol `Ca_global' can not be used when making a shared object; recompile with -fPIC
ld: final link failed: Bad value
make: *** [x86_64/libcorenrnmech.so] Error 1
```

cc: @iomaganaris  @ohm314

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -84,6 +84,11 @@ include(OpenAccHelper)
 include(ClangFormatHelper)
 
 # =============================================================================
+# ISPC should compile with --pic by default
+# =============================================================================
+set(CMAKE_ISPC_FLAGS "--pic ${CMAKE_ISPC_FLAGS}")
+
+# =============================================================================
 # Common build options
 # =============================================================================
 # build mod files for coreneuron

